remixer_storage/src/main/java/com/google/android/libraries/remixer/storage/json/SupportedDataType.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    StoredVariable<Integer> fromRemixerItem(RemixerItem item) {
+      if (item instanceof Variable) {
+        Variable var = (Variable) item;
+        if (var.getVariableType() == Integer.class
+            && var.getLayoutId() != R.layout.color_list_variable_widget) {
+          StoredVariable<Integer> storage = new StoredVariable<>();
+          storage.dataType = NUMBER.getDataTypeSerializableString();
+          storage.selectedValue = (Integer) var.getSelectedValue();
+          if (var instanceof RangeVariable) {
+            RangeVariable range = (RangeVariable) var;
+            storage.minValue = range.getMinValue();
+            storage.maxValue = range.getMaxValue();
+            storage.increment = range.getIncrement();
+          }
+          if (var instanceof ItemListVariable) {
+            storage.possibleValues = ((ItemListVariable<Integer>) var).getValueList();
+          }
+          return storage;
+        }
+      }
+      throw new IllegalArgumentException(
+          "Passed an incompatible object to convert to StoredVariable<Integer>");
+    }
+  }),
+  STRING("string", new ValueConverter<String>("string") {
+    @Override
+    public String parseValue(JsonElement element) {
+      return element.getAsString();
+    }
+
+    @Override
+    JsonElement valueToJson(String value) {
+      return new JsonPrimitive(value);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    StoredVariable<String> fromRemixerItem(RemixerItem item) {
+      if (item instanceof Variable) {
+        Variable var = (Variable) item;
+        if (var.getVariableType() == String.class) {
+          StoredVariable<String> storage = new StoredVariable<>();
+          storage.dataType = STRING.getDataTypeSerializableString();
+          storage.selectedValue = (String) var.getSelectedValue();
+          if (var instanceof ItemListVariable) {
+            storage.possibleValues = ((ItemListVariable<String>) var).getValueList();
+          }
+          return storage;
+        }
+      }
+      throw new IllegalArgumentException(
+          "Passed an incompatible object to convert to StoredVariable<String>");
+    }
+  }),
+  TRIGGER("trigger", new ValueConverter<Void>("trigger") {
+    @Override
+    Void parseValue(JsonElement element) {
+      return null;
+    }
+
+    @Override
+    JsonElement valueToJson(Void value) {
+      return null;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    StoredVariable<Void> fromRemixerItem(RemixerItem item) {
+      if (item instanceof Trigger) {
+        StoredVariable<Void> storage = new StoredVariable<>();
+        storage.dataType = TRIGGER.getDataTypeSerializableString();
+        return storage;
+      }
+      return null;
+    }
+  });
+
+  /**
+   * A string that represents this dataType when serialized. When deserializing it's used to
+   * identify the parsing logic to use.
+   */
```

Comment:
tiiiny nit: these 2 javadoc may be more useful on the accessor method rather than the field. Arguable.